7.0.0 fails to build on macOS with clang / libc++: `bold.cpp: error: no member named 'CellAt' in 'ftxui::Screen'` etc.

```
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/bold.cpp:23:16: error: no member named 'CellAt' in 'ftxui::Screen'
   23 |         screen.CellAt(x, y).bold = true;
      |         ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30: error: no viable conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30)' to 'SelectionStyle' (aka 'std::function<void (Pixel &)>')
   25 |     screen.SetSelectionStyle([](Cell&) {});
      |                              ^~~~~~~~~~~~
/opt/local/libexec/llvm-22/bin/../include/c++/v1/__functional/function.h:621:25: note: candidate constructor not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30)' to 'nullptr_t' (aka 'std::nullptr_t') for 1st argument
  621 |   _LIBCPP_HIDE_FROM_ABI function(nullptr_t) _NOEXCEPT {}
      |                         ^        ~~~~~~~~~
/opt/local/libexec/llvm-22/bin/../include/c++/v1/__functional/function.h:622:25: note: candidate constructor not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30)' to 'const function<void (ftxui::Pixel &)> &' for 1st argument
  622 |   _LIBCPP_HIDE_FROM_ABI function(const function&);
      |                         ^        ~~~~~~~~~~~~~~~
/opt/local/libexec/llvm-22/bin/../include/c++/v1/__functional/function.h:623:25: note: candidate constructor not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30)' to 'function<void (ftxui::Pixel &)> &&' for 1st argument
  623 |   _LIBCPP_HIDE_FROM_ABI function(function&&) _NOEXCEPT;
      |                         ^        ~~~~~~~~~~
/opt/local/libexec/llvm-22/bin/../include/c++/v1/__functional/function.h:625:25: note: candidate template ignored: requirement 'std::integral_constant<bool, false>::value' was not satisfied [with _Fp = (lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30)]
  625 |   _LIBCPP_HIDE_FROM_ABI function(_Fp);
      |                         ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30: note: candidate function
   25 |     screen.SetSelectionStyle([](Cell&) {});
      |                              ^
/opt/local/include/ftxui/screen/screen.hpp:74:41: note: passing argument to parameter 'decorator' here
   74 |   void SetSelectionStyle(SelectionStyle decorator);
      |                                         ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:39:7: error: no matching function for call to object of type 'const SelectionStyle' (aka 'const std::function<void (Pixel &)>')
   39 |       old_style(pixel);
      |       ^~~~~~~~~
/opt/local/libexec/llvm-22/bin/../include/c++/v1/__functional/function.h:671:29: note: candidate function not viable: no known conversion from 'Cell' to 'ftxui::Pixel &' for 1st argument
  671 |   _LIBCPP_HIDE_FROM_ABI _Rp operator()(_ArgTypes...) const;
      |                             ^          ~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:42:30: error: no viable conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:38:22)' to 'SelectionStyle' (aka 'std::function<void (Pixel &)>')
   42 |     screen.SetSelectionStyle(new_style);
      |                              ^~~~~~~~~
/opt/local/libexec/llvm-22/bin/../include/c++/v1/__functional/function.h:621:25: note: candidate constructor not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:38:22)' to 'nullptr_t' (aka 'std::nullptr_t') for 1st argument
  621 |   _LIBCPP_HIDE_FROM_ABI function(nullptr_t) _NOEXCEPT {}
      |                         ^        ~~~~~~~~~
/opt/local/libexec/llvm-22/bin/../include/c++/v1/__functional/function.h:622:25: note: candidate constructor not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:38:22)' to 'const function<void (ftxui::Pixel &)> &' for 1st argument
  622 |   _LIBCPP_HIDE_FROM_ABI function(const function&);
      |                         ^        ~~~~~~~~~~~~~~~
/opt/local/libexec/llvm-22/bin/../include/c++/v1/__functional/function.h:623:25: note: candidate constructor not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:38:22)' to 'function<void (ftxui::Pixel &)> &&' for 1st argument
  623 |   _LIBCPP_HIDE_FROM_ABI function(function&&) _NOEXCEPT;
      |                         ^        ~~~~~~~~~~
/opt/local/libexec/llvm-22/bin/../include/c++/v1/__functional/function.h:625:25: note: candidate template ignored: requirement 'std::integral_constant<bool, false>::value' was not satisfied [with _Fp = (lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:38:22)]
  625 |   _LIBCPP_HIDE_FROM_ABI function(_Fp);
      |                         ^
/opt/local/include/ftxui/screen/screen.hpp:74:41: note: passing argument to parameter 'decorator' here
   74 |   void SetSelectionStyle(SelectionStyle decorator);
      |                                         ^
[ 21%] Building CXX object CMakeFiles/dom.dir/src/ftxui/dom/color.cpp.o
/opt/local/bin/clang++-mp-22 -DIS_FTXUI_DOM_IMPL=1 -I/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src -isystem /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/include -pipe -Os -DNDEBUG -I/opt/local/include -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.15 -fPIC -MD -MT CMakeFiles/dom.dir/src/ftxui/dom/color.cpp.o -MF CMakeFiles/dom.dir/src/ftxui/dom/color.cpp.o.d -o CMakeFiles/dom.dir/src/ftxui/dom/color.cpp.o -c /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/color.cpp
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:62:10: error: no matching function for call to 'selectionStyle'
   62 |   return selectionStyle([foreground](Cell& pixel) {  //
      |          ^~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/elements.hpp:121:11: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:62:25)' to 'std::function<void (Pixel &)>' for 1st argument
  121 | Decorator selectionStyle(std::function<void(Pixel&)> style);
      |           ^              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:70:10: error: no matching function for call to 'selectionStyle'
   70 |   return selectionStyle([foreground](Cell& pixel) {  //
      |          ^~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/elements.hpp:121:11: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:70:25)' to 'std::function<void (Pixel &)>' for 1st argument
  121 | Decorator selectionStyle(std::function<void(Pixel&)> style);
      |           ^              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:96:25: error: no member named 'CellAt' in 'ftxui::Screen'
   96 |       Cell& p1 = screen.CellAt(x, box_.y_min);
      |                  ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:97:25: error: no member named 'CellAt' in 'ftxui::Screen'
   97 |       Cell& p2 = screen./opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/clear_under.cpp:25:16: error: no member named 'CellAt' in 'ftxui::Screen'
   25 |         screen.CellAt(x, y) = Cell();
      |         ~~~~~~ ^
CellAt(x, box_.y_max);
      |                  ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:104:25: error: no member named 'CellAt' in 'ftxui::Screen'
  104 |       Cell& p3 = screen.CellAt(box_.x_min, y);
      |                  ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:105:25: error: no member named 'CellAt' in 'ftxui::Screen'
  105 |       Cell& p4 = screen.CellAt(box_.x_max, y);
      |                  ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:120:16: error: no member named 'CellAt' in 'ftxui::Screen'
  120 |         screen.CellAt(x, box_.y_min).foreground_color = *foreground_color_;
      |         ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:121:16: error: no member named 'CellAt' in 'ftxui::Screen'
  121 |         screen.CellAt(x, box_.y_max).foreground_color = *foreground_color_;
      |         ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:124:16: error: no member named 'CellAt' in 'ftxui::Screen'
  124 |         screen.CellAt(box_.x_min, y).foreground_color = *foreground_color_;
      |         ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:125:16: error: no member named 'CellAt' in 'ftxui::Screen'
  125 |         screen.CellAt(box_.x_max, y).foreground_color = *foreground_color_;
      |         ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/clear_under.cpp:25:31: error: use of undeclared identifier 'Cell'
   25 |         screen.CellAt(x, y) = Cell();
      |                               ^~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/clear_under.cpp:26:16: error: no member named 'CellAt' in 'ftxui::Screen'
   26 |         screen.CellAt(x, y).character = " ";  // Consider the Cell written.
      |         ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/blink.cpp:24:16: error: no member named 'CellAt' in 'ftxui::Screen'
   24 |         screen.CellAt(x, y).blink = /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:179:12: error: no member named 'CellAt' in 'ftxui::Screen'
  179 |     screen.CellAt(box_.x_min, box_.y_min) = pixel_;
      |     ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:180:12: error: no member named 'CellAt' in 'ftxui::Screen'
  180 |     screen.CellAt(box_.x_max, box_.y_min) = pixel_;
      |     ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:181:12: error: no member named 'CellAt' in 'ftxui::Screen'
  181 |     screen.CellAt(box_.x_min, box_.y_max) = pixel_;
      |     ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:182:12: error: no member named 'CellAt' in 'ftxui::Screen'
  182 |     screen.CellAt(box_.x_max, box_.y_max) = pixel_;
      |     ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:185:14: error: no member named 'CellAt' in 'ftxui::Screen'
  185 |       screen.CellAt(x, botrx_.y_min) = piuexel_;
      |       ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:186:14: error: no member named 'CellAt' in 'ftxui::Screen'
  186 |       screen.CellAt(x, box_.y_max) = pixel_;
      |       ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:189:14: error: no member named 'CellAt' in 'ftxui::Screen'
  189 |       screen.CellAt(box_.x_min, y) = pixel_;
      |       ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:190:14: error: no member named 'CellAt' in 'ftxui::Screen'
  190 |       screen.CellAt(box_.x_max, y) = pixel_;
      |       ~~~~~~ ^
;
      |         ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/automerge.cpp:25:18: error: no member named 'CellAt' in 'ftxui::Screen'
   25 |           screen.CellAt(x, y).automerge = true;
      |           ~~~~~~ ^
In file included from /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/color.cpp:4:
In file included from /opt/local/libexec/llvm-22/bin/../include/c++/v1/memory:943:
In file included from /opt/local/libexec/llvm-22/bin/../include/c++/v1/__config:15:
/opt/local/libexec/llvm-22/bin/../include/c++/v1/__configuration/availability.h:204:4: warning: "The selected platform is no longer supported by libc++." [-W#warnings]
  204 | #  warning "The selected platform is no longer supported by libc++."
      |    ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:105:14: error: out-of-line definition of 'GetCell' does not match any declaration in 'ftxui::Canvas'
  105 | Cell Canvas::GetCell(int x, int y) const {
      |              ^~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:23:8: note: Canvas defined here
   23 | struct Canvas {
      |        ^~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:107:33: error: incompatible operand types ('Cell' and 'const Pixel')
  107 |   return (it == storage_.end()) ? Cell() : it->second.content;
      |                                 ^ ~~~~~~   ~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:115:3: error: no matching member function for call to 'DrawPoint'
  115 |   DrawPoint(x, y, value, [](Cell& /*pixel*/) {});
      |   ^~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:40:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:115:26)' to 'const Stylizer' (aka 'const std::function<void (Pixel &)>') for 4th argument
   40 |   void DrawPoint(int x, int y, bool value, const Stylizer& s);
      |        ^                                   ~~~~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:41:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:115:26)' to 'const Color' for 4th argument
   41 |   void DrawPoint(int x, int y, bool value, const Color& color);
      |        ^                                   ~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:114:14: note: candidate function not viable: requires 3 arguments, but 4 were provided
  114 | void Canvas::DrawPoint(int x, int y, bool value) {
      |              ^         ~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:124:47: error: no member named 'foreground_color' in 'ftxui::Canvas::Cell'
  124 |   DrawPoint(x, y, value, [color](Cell& p) { p.foreground_color = color; });
      |                                             ~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:124:3: error: no matching member function for call to 'DrawPoint'
  124 |   DrawPoint(x, y, value, [color](Cell& p) { p.foreground_color = color; });
      |   ^~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:40:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:124:26)' to 'const Stylizer' (aka 'const std::function<void (Pixel &)>') for 4th argument
   40 |   void DrawPoint(int x, int y, bool value, const Stylizer& s);
      |        ^                                   ~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:123:14: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:124:26)' to 'const Color' for 4th argument
  123 | void Canvas::DrawPoint(int x, int y, bool value, const Color& color) {
      |              ^                                   ~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:114:14: note: candidate function not viable: requires 3 arguments, but 4 were provided
  114 | void Canvas::DrawPoint(int x, int y, bool value) {
      |              ^         ~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:148:3: error: unknown type name 'CanvasCell'
  148 |   CanvasCell& cell = storage_[XY{x / 2, y / 4}];
      |   ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:165:3: error: unknown type name 'CanvasCell'
  165 |   CanvasCell& cell = storage_[XY{x / 2, y / 4}];
      |   ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:183:3: error: unknown type name 'CanvasCell'
  183 |   CanvasCell& cell = storage_[XY{x / 2, y / 4}];
      |   ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:199:3: error: no matching member function for call to 'DrawPointLine'
  199 |   DrawPointLine(x1, y1, x2, y2, [](Cell& /*pixel*/) {});
      |   ^~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:43:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:199:33)' to 'const Stylizer' (aka 'const std::function<void (Pixel &)>') for 5th argument
   43 |   void DrawPointLine(int x1, int y1, int x2, int y2, const Stylizer& s);
      |        ^                                             ~~~~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:44:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:199:33)' to 'const Color' for 5th argument
   44 |   void DrawPointLine(int x1, int y1, int x2, int y2, const Color& color);
      |        ^                                             ~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:198:14: note: candidate function not viable: requires 4 arguments, but 5 were provided
  198 | void Canvas::DrawPointLine(int x1, int y1, int x2, int y2) {
      |              ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:210:38: error: no member named 'foreground_color' in 'ftxui::Canvas::Cell'
  210 |                 [color](Cell& p) { p.foreground_color = color; });
      |                                    ~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:209:3: error: no matching member function for call to 'DrawPointLine'
  209 |   DrawPointLine(x1, y1, x2, y2,
      |   ^~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:43:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:210:17)' to 'const Stylizer' (aka 'const std::function<void (Pixel &)>') for 5th argument
   43 |   void DrawPointLine(int x1, int y1, int x2, int y2, const Stylizer& s);
      |        ^                                             ~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:208:14: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:210:17)' to 'const Color' for 5th argument
  208 | void Canvas::DrawPointLine(int x1, int y1, int x2, int y2, const Color& color) {
      |              ^                                             ~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:198:14: note: candidate function not viable: requires 4 arguments, but 5 were provided
  198 | void Canvas::DrawPointLine(int x1, int y1, int x2, int y2) {
      |              ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:257:3: error: no matching member function for call to 'DrawPointCircle'
  257 |   DrawPointCircle(x, y, radius, [](Cell& /*pixel*/) {});
      |   ^~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:46:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:257:33)' to 'const Stylizer' (aka 'const std::function<void (Pixel &)>') for 4th argument
   46 |   void DrawPointCircle(int x, int y, int radius, const Stylizer& s);
      |        ^                                         ~~~~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:47:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:257:33)' to 'const Color' for 4th argument
   47 |   void DrawPointCircle(int x, int y, int radius, const Color& color);
      |        ^                                         ~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:256:14: note: candidate function not viable: requires 3 arguments, but 4 were provided
  256 | void Canvas::DrawPointCircle(int x, int y, int radius) {
      |              ^               ~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:267:40: error: no member named 'foreground_color' in 'ftxui::Canvas::Cell'
  267 |                   [color](Cell& p) { p.foreground_color = color; });
      |                                      ~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:266:3: error: no matching member function for call to 'DrawPointCircle'
  266 |   DrawPointCircle(x, y, radius,
      |   ^~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:46:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:267:19)' to 'const Stylizer' (aka 'const std::function<void (Pixel &)>') for 4th argument
   46 |   void DrawPointCircle(int x, int y, int radius, const Stylizer& s);
      |        ^                                         ~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:265:14: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:267:19)' to 'const Color' for 4th argument
  265 | void Canvas::DrawPointCircle(int x, int y, int radius, const Color& color) {
      |              ^                                         ~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:256:14: note: candidate function not viable: requires 3 arguments, but 4 were provided
  256 | void Canvas::DrawPointCircle(int x, int y, int radius) {
      |              ^               ~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:284:3: error: no matching member function for call to 'DrawPointCircleFilled'
  284 |   DrawPointCircleFilled(x, y, radius, [](Cell& /*pixel*/) {});
      |   ^~~~~~~~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:49:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:284:39)' to 'const Stylizer' (aka 'const std::function<void (Pixel &)>') for 4th argument
   49 |   void DrawPointCircleFilled(int x, int y, int radius, const Stylizer& s);
      |        ^                                               ~~~~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:50:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:284:39)' to 'const Color' for 4th argument
   50 |   void DrawPointCircleFilled(int x, int y, int radius, const Color& color);
      |        ^                                               ~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:283:14: note: candidate function not viable: requires 3 arguments, but 4 were provided
  283 | void Canvas::DrawPointCircleFilled(int x, int y, int radius) {
      |              ^                     ~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:297:46: error: no member named 'foreground_color' in 'ftxui::Canvas::Cell'
  297 |                         [color](Cell& p) { p.foreground_color = color; });
      |                                            ~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:296:3: error: no matching member function for call to 'DrawPointCircleFilled'
  296 |   DrawPointCircleFilled(x, y, radius,
      |   ^~~~~~~~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:49:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:297:25)' to 'const Stylizer' (aka 'const std::function<void (Pixel &)>') for 4th argument
   49 |   void DrawPointCircleFilled(int x, int y, int radius, const Stylizer& s);
      |        ^                                               ~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:292:14: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:297:25)' to 'const Color' for 4th argument
  292 | void Canvas::DrawPointCircleFilled(int x,
      |              ^
  293 |                                    int y,
  294 |                                    int radius,
  295 |                                    const Color& color) {
      |                                    ~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:283:14: note: candidate function not viable: requires 3 arguments, but 4 were provided
  283 | void Canvas::DrawPointCircleFilled(int x, int y, int radius) {
      |              ^                     ~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:318:3: error: no matching member function for call to 'DrawPointEllipse'
  318 |   DrawPointEllipse(x, y, r1, r2, [](Cell& /*pixel*/) {});
      |   ^~~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:52:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:318:34)' to 'const Color' for 5th argument
   52 |   void DrawPointEllipse(int x, int y, int r1, int r2, const Color& color);
      |        ^                                              ~~~~~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:53:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:318:34)' to 'const Stylizer' (aka 'const std::function<void (Pixel &)>') for 5th argument
   53 |   void DrawPointEllipse(int x, int y, int r1, int r2, const Stylizer& s);
      |        ^                                              ~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:317:14: note: candidate function not viable: requires 4 arguments, but 5 were provided
  317 | void Canvas::DrawPointEllipse(int x, int y, int r1, int r2) {
      |              ^                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:333:41: error: no member named 'foreground_color' in 'ftxui::Canvas::Cell'
  333 |                    [color](Cell& p) { p.foreground_color = color; });
      |                                       ~ ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
1 warning and 20 errors generated.
1 warning and 1 error generated.
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/canvas.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/bold.cpp.o] Error 1
1 warning and 3 errors generated.
1 warning and 1 error generated.
1 warning and 5 errors generated.
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/clear_under.cpp.o] Error 1
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/blink.cpp.o] Error 1
1 warning and 1 error generated.
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/selection_style.cpp.o] Error 1
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/automerge.cpp.o] Error 1
1 warning and 16 errors generated.
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/border.cpp.o] Error 1
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/color.cpp:25:18: error: no member named 'CellAt' in 'ftxui::Screen'
   25 |           screen.CellAt(x, y).background_color = color_;
      |           ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/color.cpp:31:33: error: no member named 'CellAt' in 'ftxui::Screen'
   31 |           Color& color = screen.CellAt(x, y).background_color;
      |                          ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/color.cpp:51:18: error: no member named 'CellAt' in 'ftxui::Screen'
   51 |           screen.CellAt(x, y).foreground_color = color_;
      |           ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-4395edcd/work/FTXUI-7.0.0/src/ftxui/dom/color.cpp:57:33: error: no member named 'CellAt' in 'ftxui::Screen'
   57 |           Color& color = screen.CellAt(x, y).foreground_color;
      |                          ~~~~~~ ^
1 warning and 4 errors generated.
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/color.cpp.o] Error 1
make[2]: Leaving directory `/opt/local/var/macports/build/FTXUI-4395edcd/work/build'
make[1]: *** [CMakeFiles/dom.dir/all] Error 2
make[1]: Leaving directory `/opt/local/var/macports/build/FTXUI-4395edcd/work/build'
make: *** [all] Error 2
```

```
% sw_vers
ProductName:	Mac OS X
ProductVersion:	10.15.8
BuildVersion:	19H2036

% /opt/local/bin/clang++-mp-22 -v
clang version 22.1.8
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /opt/local/libexec/llvm-22/bin
```

P. S. The build with gcc-16 and libstdc++ on macOS 10.6.8 worked fine.

Also fails with Apple clang:
```
/usr/bin/clang++ -DIS_FTXUI_DOM_IMPL=1 -I/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src -isystem /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/include -pipe -Os -DNDEBUG -I/opt/local/include -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -std=gnu++17 -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.15 -fPIC -MD -MT CMakeFiles/dom.dir/src/ftxui/dom/border.cpp.o -MF CMakeFiles/dom.dir/src/ftxui/dom/border.cpp.o.d -o CMakeFiles/dom.dir/src/ftxui/dom/border.cpp.o -c /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp
/usr/bin/clang++ -DIS_FTXUI_DOM_IMPL=1 -I/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src -isystem /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/include -pipe -Os -DNDEBUG -I/opt/local/include -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -std=gnu++17 -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.15 -fPIC -MD -MT CMakeFiles/dom.dir/src/ftxui/dom/blink.cpp.o -MF CMakeFiles/dom.dir/src/ftxui/dom/blink.cpp.o.d -o CMakeFiles/dom.dir/src/ftxui/dom/blink.cpp.o -c /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/blink.cpp
[ 21%] Building CXX object CMakeFiles/dom.dir/src/ftxui/dom/color.cpp.o
/usr/bin/clang++ -DIS_FTXUI_DOM_IMPL=1 -I/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src -isystem /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/include -pipe -Os -DNDEBUG -I/opt/local/include -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -std=gnu++17 -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.15 -fPIC -MD -MT CMakeFiles/dom.dir/src/ftxui/dom/color.cpp.o -MF CMakeFiles/dom.dir/src/ftxui/dom/color.cpp.o.d -o CMakeFiles/dom.dir/src/ftxui/dom/color.cpp.o -c /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/color.cpp
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/automerge.cpp:25:18: error: no member named 'CellAt' in 'ftxui::Screen'
          screen.CellAt(x, y).automerge = true;
          ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/clear_under.cpp:25:16: error: no member named 'CellAt' in 'ftxui::Screen'
        screen.CellAt(x, y) = Cell();
        ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/bold.cpp:23:/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/blink.cpp:24:16: error16: error: : no member named 'CellAt' in 'ftxui::Screen'no member named 'CellAt' in 'ftxui::Screen'

        screen.CellAt(x, y).bold = true;        screen.CellAt(x, y).blink = true;

        ~~~~~~ ^
        ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/clear_under.cpp:25:31: error: use of undeclared identifier 'Cell'
        screen.CellAt(x, y) = Cell();
                              ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/clear_under.cpp:26:16: error: no member named 'CellAt' in 'ftxui::Screen'
        screen.CellAt(x, y).character = " ";  // Consider the Cell written.
        ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30: error: no viable conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30)' to 'ftxui::Screen::SelectionStyle' (aka 'function<void (ftxui::Pixel &)>')
    screen.SetSelectionStyle([](Cell&) {});
                             ^~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:96:25: error: no member named 'CellAt' in 'ftxui::Screen'
      Cell& p1 = screen.CellAt(x, box_.y_min);
                 ~~~~~~ ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/functional/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:97:25::2358:5: note: candidate constructor not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30)' to 'std::nullptr_t' (aka 'nullptr_t') for 1st argument
    function(nullptr_t) _NOEXCEPT {}
    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/functional:2359:5: note: candidate constructor not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30)' to 'const std::__1::function<void (ftxui::Pixel &)> &' for 1st argument
    function(const function&);
    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/functional:2360:5: note error: candidate constructor not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30)' to 'std::__1::function<void (ftxui::Pixel &)> &&' for 1st argument
    function(function&&) _NOEXCEPT;
    ^
: no member named 'CellAt' in 'ftxui::Screen'
      Cell& p2 = screen.CellAt(x, box_.y_max);
                 ~~~~~~ ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/functional:2362:5: note: candidate template ignored: requirement '__callable<(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30), false>::value' was not satisfied [with _Fp = (lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30)]
    function(_Fp);
    ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:25:30: note: candidate function
    screen.SetSelectionStyle([](Cell&) {});
                             ^
/opt/local/include/ftxui/screen/screen.hpp:74:41: note: passing argument to parameter 'decorator' here
  void SetSelectionStyle(SelectionStyle decorator);
                                        ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:104:25: error: no member named 'CellAt' in 'ftxui::Screen'
      Cell& p3 = screen.CellAt(box_.x_min, y);
                 ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:105:25: error: no member named 'CellAt' in 'ftxui::Screen'
      Cell& p4 = screen.CellAt(box_.x_max, y);
                 ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:120:16: error: no member named 'CellAt' in 'ftxui::Screen'
        screen.CellAt(x, box_.y_min).foreground_color = *foreground_color_;
        ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:121:16: error: no member named 'CellAt' in 'ftxui::Screen'
        screen.CellAt(x, box_.y_max).foreground_color = *foreground_color_;
        ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:124:16: error: no member named 'CellAt' in 'ftxui::Screen'
        screen.CellAt(box_.x_min, y).foreground_color = *foreground_color_;
        ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:125:16: error: no member named 'CellAt' in 'ftxui::Screen'
        screen.CellAt(box_.x_max, y).foreground_color = *foreground_color_;
        ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:179:12: error: no member named 'CellAt' in 'ftxui::Screen'
    screen.CellAt(box_.x_min, box_.y_min) = pixel_;
    ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:180:12: error: no member named 'CellAt' in 'ftxui::Screen'
    screen.CellAt(box_.x_max, box_.y_min) = pixel_;
    ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:181:12: error: no member named 'CellAt' in 'ftxui::Screen'
    screen.CellAt(box_.x_min, box_.y_max) = pixel_;
    ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:182:12: error: no member named 'CellAt' in 'ftxui::Screen'
    screen.CellAt(box_.x_max, box_.y_max) = pixel_;
    ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:185:14: error: no member named 'CellAt' in 'ftxui::Screen'
      screen.CellAt(x, box_.y_min) = pixel_;
      ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:186:14: error: no member named 'CellAt' in 'ftxui::Screen'
      screen.CellAt(x, box_.y_max) = pixel_;
      ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:189:14: error: no member named 'CellAt' in 'ftxui::Screen'
      screen.CellAt(box_.x_min, y) = pixel_;
      ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/border.cpp:190:14: error: no member named 'CellAt' in 'ftxui::Screen'
      screen.CellAt(box_.x_max, y) = pixel_;
      ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:39:7: error: no matching function for call to object of type 'const std::__1::function<void (ftxui::Pixel &)>'
      old_style(pixel);
      ^~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/functional:2410:9: note: candidate function not viable: no known conversion from 'ftxui::Cell' to 'ftxui::Pixel &' for 1st argument
    _Rp operator()(_ArgTypes...) const;
        ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:42:30: error: no viable conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:38:22)' to 'ftxui::Screen::SelectionStyle' (aka 'function<void (ftxui::Pixel &)>')
    screen.SetSelectionStyle(new_style);
                             ^~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/functional:2358:5: note: candidate constructor not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:38:22)' to 'std::nullptr_t' (aka 'nullptr_t') for 1st argument
    function(nullptr_t) _NOEXCEPT {}
    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/functional:2359:5: note: candidate constructor not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:38:22)' to 'const std::__1::function<void (ftxui::Pixel &)> &' for 1st argument
    function(const function&);
    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/functional:2360:5: note: candidate constructor not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:38:22)' to 'std::__1::function<void (ftxui::Pixel &)> &&' for 1st argument
    function(function&&) _NOEXCEPT;
    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/functional:2362:5: note: candidate template ignored: requirement '__callable<(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:38:22), false>::value' was not satisfied [with _Fp = (lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:38:22)]
    function(_Fp);
    ^
/opt/local/include/ftxui/screen/screen.hpp:74:41: note: passing argument to parameter 'decorator' here
  void SetSelectionStyle(SelectionStyle decorator);
                                        ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:62:10: error: no matching function for call to 'selectionStyle'
  return selectionStyle([foreground](Cell& pixel) {  //
         ^~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/elements.hpp:121:11: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:62:25)' to 'std::function<void (Pixel &)>' for 1st argument
Decorator selectionStyle(std::function<void(Pixel&)> style);
          ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:70:10: error: no matching function for call to 'selectionStyle'
  return selectionStyle([foreground](Cell& pixel) {  //
         ^~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/elements.hpp:121:11: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/selection_style.cpp:70:25)' to 'std::function<void (Pixel &)>' for 1st argument
Decorator selectionStyle(std::function<void(Pixel&)> style);
          ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:105:14: error: out-of-line definition of 'GetCell' does not match any declaration in 'ftxui::Canvas'
Cell Canvas::GetCell(int x, int y) const {
             ^~~~~~~
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:107:33: error: incompatible operand types ('ftxui::Canvas::Cell' and 'const ftxui::Pixel')
  return (it == storage_.end()) ? Cell() : it->second.content;
                                ^ ~~~~~~   ~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:115:3: error: no matching member function for call to 'DrawPoint'
  DrawPoint(x, y, value, [](Cell& /*pixel*/) {});
  ^~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:40:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:115:26)' to 'const ftxui::Canvas::Stylizer' (aka 'const function<void (ftxui::Pixel &)>') for 4th argument
  void DrawPoint(int x, int y, bool value, const Stylizer& s);
       ^
/opt/local/include/ftxui/dom/canvas.hpp:41:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:115:26)' to 'const ftxui::Color' for 4th argument
  void DrawPoint(int x, int y, bool value, const Color& color);
       ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:114:14: note: candidate function not viable: requires 3 arguments, but 4 were provided
void Canvas::DrawPoint(int x, int y, bool value) {
             ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:124:47: error: no member named 'foreground_color' in 'ftxui::Canvas::Cell'
  DrawPoint(x, y, value, [color](Cell& p) { p.foreground_color = color; });
                                            ~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:124:3: error: no matching member function for call to 'DrawPoint'
  DrawPoint(x, y, value, [color](Cell& p) { p.foreground_color = color; });
  ^~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:40:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:124:26)' to 'const ftxui::Canvas::Stylizer' (aka 'const function<void (ftxui::Pixel &)>') for 4th argument
  void DrawPoint(int x, int y, bool value, const Stylizer& s);
       ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:123:14: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:124:26)' to 'const ftxui::Color' for 4th argument
void Canvas::DrawPoint(int x, int y, bool value, const Color& color) {
             ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:114:14: note: candidate function not viable: requires 3 arguments, but 4 were provided
void Canvas::DrawPoint(int x, int y, bool value) {
             ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:148:3: error: unknown type name 'CanvasCell'
  CanvasCell& cell = storage_[XY{x / 2, y / 4}];
  ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:165:3: error: unknown type name 'CanvasCell'
  CanvasCell& cell = storage_[XY{x / 2, y / 4}];
  ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:183:3: error: unknown type name 'CanvasCell'
  CanvasCell& cell = storage_[XY{x / 2, y / 4}];
  ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:199:3: error: no matching member function for call to 'DrawPointLine'
  DrawPointLine(x1, y1, x2, y2, [](Cell& /*pixel*/) {});
  ^~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:43:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:199:33)' to 'const ftxui::Canvas::Stylizer' (aka 'const function<void (ftxui::Pixel &)>') for 5th argument
  void DrawPointLine(int x1, int y1, int x2, int y2, const Stylizer& s);
       ^
/opt/local/include/ftxui/dom/canvas.hpp:44:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:199:33)' to 'const ftxui::Color' for 5th argument
  void DrawPointLine(int x1, int y1, int x2, int y2, const Color& color);
       ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:198:14: note: candidate function not viable: requires 4 arguments, but 5 were provided
void Canvas::DrawPointLine(int x1, int y1, int x2, int y2) {
             ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:210:38: error: no member named 'foreground_color' in 'ftxui::Canvas::Cell'
                [color](Cell& p) { p.foreground_color = color; });
                                   ~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:209:3: error: no matching member function for call to 'DrawPointLine'
  DrawPointLine(x1, y1, x2, y2,
  ^~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:43:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:210:17)' to 'const ftxui::Canvas::Stylizer' (aka 'const function<void (ftxui::Pixel &)>') for 5th argument
  void DrawPointLine(int x1, int y1, int x2, int y2, const Stylizer& s);
       ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:208:14: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:210:17)' to 'const ftxui::Color' for 5th argument
void Canvas::DrawPointLine(int x1, int y1, int x2, int y2, const Color& color) {
             ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:198:14: note: candidate function not viable: requires 4 arguments, but 5 were provided
void Canvas::DrawPointLine(int x1, int y1, int x2, int y2) {
             ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:257:3: error: no matching member function for call to 'DrawPointCircle'
  DrawPointCircle(x, y, radius, [](Cell& /*pixel*/) {});
  ^~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:46:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:257:33)' to 'const ftxui::Canvas::Stylizer' (aka 'const function<void (ftxui::Pixel &)>') for 4th argument
  void DrawPointCircle(int x, int y, int radius, const Stylizer& s);
       ^
/opt/local/include/ftxui/dom/canvas.hpp:47:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:257:33)' to 'const ftxui::Color' for 4th argument
  void DrawPointCircle(int x, int y, int radius, const Color& color);
       ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:256:14: note: candidate function not viable: requires 3 arguments, but 4 were provided
void Canvas::DrawPointCircle(int x, int y, int radius) {
             ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:267:40: error: no member named 'foreground_color' in 'ftxui::Canvas::Cell'
                  [color](Cell& p) { p.foreground_color = color; });
                                     ~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:266:3: error: no matching member function for call to 'DrawPointCircle'
  DrawPointCircle(x, y, radius,
  ^~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:46:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:267:19)' to 'const ftxui::Canvas::Stylizer' (aka 'const function<void (ftxui::Pixel &)>') for 4th argument
  void DrawPointCircle(int x, int y, int radius, const Stylizer& s);
       ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:265:14: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:267:19)' to 'const ftxui::Color' for 4th argument
void Canvas::DrawPointCircle(int x, int y, int radius, const Color& color) {
             ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:256:14: note: candidate function not viable: requires 3 arguments, but 4 were provided
void Canvas::DrawPointCircle(int x, int y, int radius) {
             ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:284:3: error: no matching member function for call to 'DrawPointCircleFilled'
  DrawPointCircleFilled(x, y, radius, [](Cell& /*pixel*/) {});
  ^~~~~~~~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:49:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:284:39)' to 'const ftxui::Canvas::Stylizer' (aka 'const function<void (ftxui::Pixel &)>') for 4th argument
  void DrawPointCircleFilled(int x, int y, int radius, const Stylizer& s);
       ^
/opt/local/include/ftxui/dom/canvas.hpp:50:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:284:39)' to 'const ftxui::Color' for 4th argument
  void DrawPointCircleFilled(int x, int y, int radius, const Color& color);
       ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:283:14: note: candidate function not viable: requires 3 arguments, but 4 were provided
void Canvas::DrawPointCircleFilled(int x, int y, int radius) {
             ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:297:46: error: no member named 'foreground_color' in 'ftxui::Canvas::Cell'
                        [color](Cell& p) { p.foreground_color = color; });
                                           ~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:296:3: error: no matching member function for call to 'DrawPointCircleFilled'
  DrawPointCircleFilled(x, y, radius,
  ^~~~~~~~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:49:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:297:25)' to 'const ftxui::Canvas::Stylizer' (aka 'const function<void (ftxui::Pixel &)>') for 4th argument
  void DrawPointCircleFilled(int x, int y, int radius, const Stylizer& s);
       ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:292:14: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:297:25)' to 'const ftxui::Color' for 4th argument
void Canvas::DrawPointCircleFilled(int x,
             ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:283:14: note: candidate function not viable: requires 3 arguments, but 4 were provided
void Canvas::DrawPointCircleFilled(int x, int y, int radius) {
             ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:318:3: error: no matching member function for call to 'DrawPointEllipse'
  DrawPointEllipse(x, y, r1, r2, [](Cell& /*pixel*/) {});
  ^~~~~~~~~~~~~~~~
/opt/local/include/ftxui/dom/canvas.hpp:52:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:318:34)' to 'const ftxui::Color' for 5th argument
  void DrawPointEllipse(int x, int y, int r1, int r2, const Color& color);
       ^
/opt/local/include/ftxui/dom/canvas.hpp:53:8: note: candidate function not viable: no known conversion from '(lambda at /opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:318:34)' to 'const ftxui::Canvas::Stylizer' (aka 'const function<void (ftxui::Pixel &)>') for 5th argument
  void DrawPointEllipse(int x, int y, int r1, int r2, const Stylizer& s);
       ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:317:14: note: candidate function not viable: requires 4 arguments, but 5 were provided
void Canvas::DrawPointEllipse(int x, int y, int r1, int r2) {
             ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/canvas.cpp:333:41: error: no member named 'foreground_color' in 'ftxui::Canvas::Cell'
                   [color](Cell& p) { p.foreground_color = color; });
                                      ~ ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/canvas.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
1 error generated.
1 error generated.
3 errors generated.
1 error generated.
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/automerge.cpp.o] Error 1
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/blink.cpp.o] Error 1
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/clear_under.cpp.o] Error 1
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/bold.cpp.o] Error 1
5 errors generated.
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/selection_style.cpp.o] Error 1
16 errors generated.
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/border.cpp.o] Error 1
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/color.cpp:25:18: error: no member named 'CellAt' in 'ftxui::Screen'
          screen.CellAt(x, y).background_color = color_;
          ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/color.cpp:31:33: error: no member named 'CellAt' in 'ftxui::Screen'
          Color& color = screen.CellAt(x, y).background_color;
                         ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/color.cpp:51:18: error: no member named 'CellAt' in 'ftxui::Screen'
          screen.CellAt(x, y).foreground_color = color_;
          ~~~~~~ ^
/opt/local/var/macports/build/FTXUI-2275dbc7/work/FTXUI-7.0.0/src/ftxui/dom/color.cpp:57:33: error: no member named 'CellAt' in 'ftxui::Screen'
          Color& color = screen.CellAt(x, y).foreground_color;
                         ~~~~~~ ^
4 errors generated.
make[2]: *** [CMakeFiles/dom.dir/src/ftxui/dom/color.cpp.o] Error 1
make[2]: Leaving directory `/opt/local/var/macports/build/FTXUI-2275dbc7/work/build'
make[1]: *** [CMakeFiles/dom.dir/all] Error 2
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,13 @@ Changelog
 Next
 ====
 
+### Build
+- Bugfix: Fix build failure when an older FTXUI is installed in a system
+  include path (e.g. MacPorts upgrade). A CMake deduplication quirk was
+  promoting the project's own `-I include/` to `-isystem`, causing package
+  managers' `-I/opt/local/include` (which may contain stale headers) to
+  win. See #1299, #1300.
+
 7.0.0 (2026-06-13)
 ------------------
 

--- a/cmake/ftxui_set_options.cmake
+++ b/cmake/ftxui_set_options.cmake
@@ -27,19 +27,34 @@ function(ftxui_set_options library)
       PROPERTIES CXX_CLANG_TIDY "${CLANG_TIDY_EXE};-warnings-as-errors=*"
     )
 
-    # By using "PUBLIC" as opposed to "SYSTEM INTERFACE", the compiler warning
-    # are enforced on the headers. This is behind "FTXUI_CLANG_TIDY", so that it
-    # applies only when developing FTXUI and on the CI. User's of the library
-    # get only the SYSTEM INTERFACE instead.
+    # By using "PUBLIC" as opposed to "INTERFACE", the compiler warnings are
+    # enforced on the headers. This is behind "FTXUI_CLANG_TIDY", so that it
+    # applies only when developing FTXUI and on the CI. Users of the library
+    # get only the plain INTERFACE instead.
     target_include_directories(${library}
       PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     )
   else()
-    target_include_directories(${library} SYSTEM
-      INTERFACE
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-    )
+    if (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+      # Building FTXUI as a standalone project.  Do NOT mark as SYSTEM: if
+      # a sibling FTXUI target (e.g. dom depending on screen) inherits this
+      # path via INTERFACE_SYSTEM_INCLUDE_DIRECTORIES, CMake demotes the
+      # private -I to -isystem.  That path is then searched *after* any -I
+      # added by a package manager (e.g. MacPorts' -I/opt/local/include),
+      # letting a stale installed FTXUI shadow the sources being built.
+      target_include_directories(${library}
+        INTERFACE
+          $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+      )
+    else()
+      # Consumed via add_subdirectory / FetchContent: keep SYSTEM so that
+      # consumers do not receive warnings from FTXUI headers.
+      target_include_directories(${library} SYSTEM
+        INTERFACE
+          $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+      )
+    endif()
   endif()
 
   target_include_directories(${library} SYSTEM

--- a/include/ftxui/screen/screen.hpp
+++ b/include/ftxui/screen/screen.hpp
@@ -34,6 +34,10 @@ class FTXUI_EXPORT(SCREEN) Screen : public Surface {
   // Destructor:
   ~Screen() override = default;
 
+  // Copy:
+  Screen(const Screen&) = default;
+  Screen& operator=(const Screen&) = default;
+
   std::string ToString() const;
   void ToString(std::string& ss) const;
 

--- a/include/ftxui/screen/surface.hpp
+++ b/include/ftxui/screen/surface.hpp
@@ -27,6 +27,10 @@ class FTXUI_EXPORT(SCREEN) Surface {
   // Destructor:
   virtual ~Surface() = default;
 
+  // Copy:
+  Surface(const Surface&) = default;
+  Surface& operator=(const Surface&) = default;
+
   // Access a character in the grid at a given position.
   std::string& at(int x, int y);
   const std::string& at(int x, int y) const;

